### DISCUSSION TITLE
Android cannot generate p2sh address

### DISCUSSION THREAD

**Discussion by justresponse** [trustwallet/wallet-core#3616] [Q&A]
I am an android developer. Is there a method like let address = BitcoinAddress.compatibleAddress(publicKey: publicKey, prefix: CoinType.bitcoin.p2shPrefix) that allows me to generate a p2sh address?

**Comment by satoshiotomakan** [comment]
Hi @justresponse, it's technically possible to create a p2sh address, but not recommended.
We do not support sending funds from p2sh addresses, only p2pkh, segwit and taproot.

**Reply by justresponse** [reply]
Does Android have a way to generate an address like iOS does? For example: let address = BitcoinAddress.compatibleAddress(publicKey: publicKey, prefix: CoinType.bitcoin.p2shPrefix)
Because I couldn't find a similar method in the latest jar package, please provide it, thank you.

**Reply by lamafab** [reply]
I assume you're looking for this (?):

```kotlin
actual class BitcoinAddress private constructor(
    private val nativeHandle: Long,
) {
    // ...

    actual companion object {
        // ...

        @JvmStatic
        @JvmName("createWithPublicKey")
        private external fun createWithPublicKey(publicKey: PublicKey, prefix: UByte): Long
    }
}
```
